verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -42,25 +51,177 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
     return status;
   }
 
-  // TODO(fangism): parse a subcommand-specific flag for replacement option
-  // See documentation for 'replacement' arg of the library function:
-  verilog::StripVerilogComments(source_contents, &outs, ' ');
+  char replace_char;
+  if (args.size() == 1) {
+    replace_char = ' ';
+  } else if (args.size() == 2) {
+    absl::string_view replace_str(args[1]);
+    if (replace_str.empty())
+      replace_char = '\0';
+    else if (replace_str.length() == 1)
+      replace_char = replace_str[0];
+    else
+      return absl::InvalidArgumentError(
+          "Replacement must be a single character.");
+  } else {
+    return absl::InvalidArgumentError("Too many arguments.");
+  }
+
+  verilog::StripVerilogComments(source_contents, &outs, replace_char);
 
   return absl::OkStatus();
 }
 
-static const std::pair<absl::string_view, SubcommandEntry> kCommands[] = {
-    {"strip-comments",  //
-     {&StripComments,   //
-      R"(strip-comments file
+static absl::Status PreprocessSingleFile(const char* source_file, std::istream&,
```

Comment:
What is the `std::istream&` intended to be used for ? Looks like it can be removed.

Also the second `std::ostream&` ? It looks like the second ostream is the `message_stream`; we can use that instead of `std::cerr` below.

Parameters that we don't need: let's not include them, as it just gets confusing.

Reply:
Check my previous comment https://github.com/chipsalliance/verible/pull/1372#discussion_r960428463

Reply:
PreprocessSingleFile() is never called from the subcommand. You only register `MultipleCU()` and `GenerateVarints()` with the subocmmands so that you have to adhere to that function signature.
But only from within these functions, you then call `PreprocessSingleFile()`, but you don't need the extra parameters here.